genrl/core/actor_critic.py

```diff
@@ -190,6 +340,8 @@ def get_value(self, state: torch.Tensor, mode="first") -> torch.Tensor:
             values (:obj:`list`): List of values as estimated by each individual critic
         """
         state = torch.as_tensor(state).float()
+        x = self.get_features(state[:, :, :-1])
+        state = torch.cat([x, state[:, :, -1].unsqueeze(-1)], dim=-1)
```

Comment:
What's with the weird shaping? Can you use `torch.reshape` instead?

Reply:
Couldn't comment on below but `mode == "min"` seems to be missing

Reply:
+1 shaping stuff

Atleast add a comment wherever you're doing indexing so that its easy to debug later on and we dont have to chase you.